aten/src/ATen/native/Pow.cpp

```diff
@@ -89,24 +89,32 @@ Tensor& float_power_out(Tensor& result, const Tensor& base, const Tensor& exp) {
 }
 
 Tensor& float_power_out(Tensor& result, const Tensor& base, Scalar exp) {
-  return at::float_power_out(result, base, c10::scalar_to_tensor(exp, base.device()));
+  auto dtype = (at::isComplexType(base.scalar_type()) || exp.isComplex()) ? at::kComplexDouble : at::kDouble;
+  exp = dtype == at::kComplexDouble ? exp.toComplexDouble() : exp.toDouble();
+  return at::pow_out(result, base.to(dtype), exp);
 }
 
 Tensor& float_power_out(Tensor& result, Scalar base, const Tensor& exp) {
-  return at::float_power_out(result, c10::scalar_to_tensor(base, exp.device()), exp);
-}
-
-Tensor float_power(const Tensor& base, const Tensor& exp) {
-  auto dtype = (at::isComplexType(base.scalar_type()) || at::isComplexType(exp.scalar_type())) ? at::kComplexDouble : at::kDouble;
-  return at::pow(base.to(dtype), exp.to(dtype));
+  auto dtype = (at::isComplexType(exp.scalar_type()) || base.isComplex()) ? at::kComplexDouble : at::kDouble;
+  base = dtype == at::kComplexDouble ? base.toComplexDouble() : base.toDouble();
```

Comment:
Ideally we'd add an output checking helper (or maybe one exists that we can reuse) here. Otherwise if result has the wrong dtype or the wrong device type the error will be throw in pow, and will not correctly identify float_power as the source of the issue. This could be confusing to users.